internal/statestore/backfill.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statestore
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/gomodule/redigo/redis"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+// CreateBackfill creates a new Backfill in the state storage. If the id already exists, it will be overwritten.
```

Comment:
> If the id already exists, it will be overwritten.

This sounds dangerous. What are the cons of just having the code that creates the entry also create the ID for it?

Reply:
We were supposing to keep State Store simple as it was, and generate this ID on Frontend. I will sent a PR soon, it would be similar to `CreateTicket`.

Reply:
The xids algorithm used to create the ids ensures that they are unique with no system wide synchronization.  Calling clients are forbidden from choosing an id during create.  So no conflicts will occur.

Reply:
👍